### PR TITLE
fix: add common error handling for missing or wrong username

### PR DESCRIPTION
**Pull request by erdahuja**
**What**:
1. Added username validation in inquirer prompt
2. Add validation for wrong username while fetching info from github.

issue #6 

**Why**:
1. More informative errors.
2. Better usability of package .

**How**:

1. Added validation inquirer prompt.
2. throwing error for no username in getUserInfo for github

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Ready to be merged 
- [ ] Added myself to contributors table

### CODE REVIEW

File: src/repo/__tests__/github.js [hunk 1]
```diff
@@ -65,6 +65,10 @@ test('handle errors', async () => {
   await rejects(getUserInfo('nodisplayname'))
 })
 
+test('Throw error when no username is provided', async () => {
+    expect(await getUserInfo()).toThrow()
```
Comment (erdahuja) [inline src/repo/__tests__/github.js]: Can anyone please help me with:

```suggestion
    expect(await getUserInfo()).toThrow()
```

Even though the function throws error, the test still fails as if throwing error shouldn't pass the test. @Berkmann18 @kentcdodds

Comment (sinchang) [reply src/repo/__tests__/github.js]: you can try about that.
```
test('Throw error when no username is provided', async () => {
    try {
      await getUserInfo();
    } catch (e) {
      expect(e.message).toEqual('No login when adding a contributor. Please specify a username.');
    }
})
```

File: src/repo/github.js [hunk 2]
```diff
@@ -60,6 +60,12 @@ const getUserInfo = function(username, hostname, optionalPrivateToken) {
     hostname = 'https://github.com'
   }
 
+  if (!username) {
+    throw new Error(
+      `No login when adding a contributor. Please specify a username.`,
```
Comment (Berkmann18) [inline src/repo/github.js]: The CI is failing here, please make sure your test passes locally.

**Comment by Berkmann18** [comment]
Make sure you follow the [Angular commit convention](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines) as explained in the [contribution guidelines](https://allcontributors.org/docs/en/project/contribute#committing-to-an-all-contributors-repo).

**Comment by erdahuja** [comment]
> Make sure you follow the [Angular commit convention](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines) as explained in the [contribution guidelines](https://allcontributors.org/docs/en/project/contribute#committing-to-an-all-contributors-repo).

Okay, thanks. I'll rebase and update once review process is completed. I think it's done now. since my doubt is resolved. Thanks!

File: src/repo/__tests__/github.js [hunk 3]
```diff
@@ -65,6 +65,10 @@ test('handle errors', async () => {
   await rejects(getUserInfo('nodisplayname'))
 })
 
+test('Throw error when no username is provided', () => {
+    expect(getUserInfo).toThrow('No login when adding a contributor. Please specify a username.')
```
Comment (erdahuja) [inline src/repo/__tests__/github.js]: @sinchang I did it like this. Re read documentation. Actually expect should call the function not us.

Comment (erdahuja) [reply src/repo/__tests__/github.js]: @Berkmann18 friendly ping

**Pull request merged**